src/main/java/org/mule/extension/http/internal/request/RequestUrlConfiguration.java

```diff
@@ -21,4 +21,25 @@
   public String getBasePath() {
     return basePath;
   }
+
+  public static class Builder {
+
+    private String basePath;
+
+    public Builder withBasePath(String basePath) {
+      this.basePath = basePath;
+      return this;
+    }
+
+    public RequestUrlConfiguration build() {
+      RequestUrlConfiguration config = new RequestUrlConfiguration();
+      config.basePath = this.basePath;
+      return config;
+    }
+
+    public static Builder newInstance() {
+      return new Builder();
+    }
+
+  }
```

Comment:
instead of having a `newInstance` static method inside the Builder, add a `builder` static method within the RequestUrlConfiguration class. 
Same for the rest of the Builders